WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteSourceManager.kt

```diff
@@ -93,6 +93,13 @@ class MySiteSourceManager @Inject constructor(
         quickStartCardSource.refresh()
     }
 
+    private fun <X> LiveData<X>.addDistinctUntilChangedIfNeeded(): LiveData<X> =
```

Comment:
1. Suggestion (💡): What about moving this extension outside on this class, but still part of this file, then make it public and share that with with `MySiteViewModel` in order to remove the duplication and as such make it easy to update in the future, or else to not forget to update one if the other is updated. Wdyt?
2. Question (❓): Why do you need the `Transformations.distinctUntilChanged(this)`, wouldn't using the `distinctUntilChanged()` extension directly work at it used to work before?
3. Suggestion (💡): Also as part of the above suggestion, if I may, I want to suggest the following rewrite:

`MySiteSourceManager.kt`
```
fun <X> LiveData<X>.addDistinctUntilChangedIfNeeded(isNeeded: Boolean): LiveData<X> {
    return if (isNeeded) {
        distinctUntilChanged()
    } else {
        this
    }
}

    fun build(coroutineScope: CoroutineScope, siteLocalId: Int?): List<LiveData<out PartialState>> {
        return if (siteLocalId != null) {
            mySiteSources.map { source ->
                source.build(coroutineScope, siteLocalId)
                        .addDistinctUntilChangedIfNeeded(!mySiteDashboardPhase2FeatureConfig.isEnabled())
            }
        } else {
            mySiteSources.filterIsInstance(SiteIndependentSource::class.java).map { source ->
                source.build(coroutineScope)
                        .addDistinctUntilChangedIfNeeded(!mySiteDashboardPhase2FeatureConfig.isEnabled())
            }
        }
    }
```

`MySiteViewModel.kt`
```
    val state: LiveData<MySiteUiState> =
            selectedSiteRepository.siteSelected.switchMap { siteLocalId ->
                val result = MediatorLiveData<SiteIdToState>()
                for (newSource in mySiteSourceManager.build(viewModelScope, siteLocalId)) {
                    result.addSource(newSource) { partialState ->
                        if (partialState != null) {
                            result.value = (result.value ?: SiteIdToState(siteLocalId)).update(partialState)
                        }
                    }
                }
                // We want to filter out the empty state where we have a site ID but site object is missing.
                // Without this check there is an emission of a NoSites state even if we have the site
                result.filter { it.siteId == null || it.state.site != null }.map { it.state }
            }.addDistinctUntilChangedIfNeeded(!mySiteDashboardPhase2FeatureConfig.isEnabled())
```

Reply:
👋 @zwarm !

I say, let's keep it, since it doesn't affect `develop` and keep testing it for a few days to figure out if we can work with it, or find any issues, that is in addition to the dimming that happens. Then, we can try and figure out if we can fix the dimming as well and another solution or an update to this current solution. Wdyt?

Reply:
I added the `Do not merge` until after our discussion

Reply:
Committed to the functionality and addressed my original concerns by creating a single extension function. Incorporated @ParaskP7 idea of passing the featureFlag value to the extension directly. 👍  In 1ed958f. Thus I am removing the Do not merge label.

Reply:
Great, as discussed, let's revisit that but for now go with it, thank you @zwarm ! 🚀